internal/core/algorithm/ngt/ngt_test.go

```diff
@@ -0,0 +1,3551 @@
+					{ID: uint32(1), Distance: 3},
+				},
+			},
+		},
+
+		// other cases
+		{
+			name: "return nothing if the search dimension is less than the inserted vector",
+			args: args{
+				vec:     []float32{0, 1, 2, 3, 4, 5, 6, 7},
+				size:    5,
+				epsilon: 0,
+				radius:  0,
+			},
+			fields: fields{
+				inMemory:            false,
+				bulkInsertChunkSize: 100,
+				dimension:           9,
+				objectType:          Uint8,
+				radius:              float32(-1.0),
+				epsilon:             float32(0.01),
+			},
+			createFunc: func(t *testing.T, fields fields) (NGT, error) {
+				t.Helper()
+				vec := []float32{0, 1, 2, 3, 4, 5, 6, 7, 8}
+
+				return insertCreateFunc(t, fields, [][]float32{vec}, 1)
+			},
+			want: want{
+				err: errors.New("incompatible dimension size detected\trequested: 8,\tconfigured: 9"),
+			},
+		},
+		{
+			name: "return nothing if the search dimension is more than the inserted vector",
+			args: args{
+				vec:     []float32{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+				size:    5,
+				epsilon: 0,
+				radius:  0,
+			},
+			fields: fields{
+				inMemory:            false,
+				bulkInsertChunkSize: 100,
+				dimension:           9,
+				objectType:          Uint8,
+				radius:              float32(-1.0),
+				epsilon:             float32(0.01),
+			},
+			createFunc: func(t *testing.T, fields fields) (NGT, error) {
+				t.Helper()
+				vec := []float32{0, 1, 2, 3, 4, 5, 6, 7, 8}
+
+				return insertCreateFunc(t, fields, [][]float32{vec}, 1)
+			},
+			want: want{
+				err: errors.New("incompatible dimension size detected\trequested: 10,\tconfigured: 9"),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		test := tc
+		t.Run(test.name, func(tt *testing.T) {
+			tt.Parallel()
+			defer goleak.VerifyNone(tt, goleak.IgnoreCurrent())
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc == nil {
+				test.afterFunc = defaultAfterFunc
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+			if test.createFunc == nil {
+				test.createFunc = defaultCreateFunc
+			}
+
+			n, err := test.createFunc(tt, test.fields)
+			if err != nil {
+				tt.Fatal(err)
+			}
+
+			got, err := n.Search(test.args.vec, test.args.size, test.args.epsilon, test.args.radius)
+			if err := test.checkFunc(test.want, got, n, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+
+			if err := test.afterFunc(tt, n); err != nil {
+				tt.Error(err)
+			}
+		})
+	}
+}
+
+func Test_ngt_Insert(t *testing.T) {
+	type args struct {
+		vec []float32
+	}
+	type fields struct {
+		idxPath             string
+		inMemory            bool
+		bulkInsertChunkSize int
+		dimension           int
+		objectType          objectType
+		radius              float32
+		epsilon             float32
+		poolSize            uint32
+		// prop                C.NGTProperty
+		// ebuf                C.NGTError
+		// index               C.NGTIndex
+		// ospace              C.NGTObjectSpace
```

Comment:
Are they needed?

Reply:
They are c dependencies, even it is needed, I think it is better to comment out them